components/sidebar/sidebar_category/sidebar_category.tsx

```diff
@@ -157,33 +153,32 @@ export default class SidebarCategory extends React.PureComponent<Props, State> {
 
         if (category.type === CategoryTypes.DIRECT_MESSAGES) {
             return draggingState.type === DraggingStateTypes.CHANNEL;
-        } else if (category.type === CategoryTypes.PUBLIC || category.type === CategoryTypes.PRIVATE) {
+        } else if (category.type === CategoryTypes.CHANNELS) {
             return draggingState.type === DraggingStateTypes.DM;
         }
 
         return false;
     }
 
     render() {
-        const {category, categoryIndex, isCollapsed, draggingState, channels} = this.props;
-        const {isNewCategory} = this.state;
+        const {category, categoryIndex, isCollapsed, draggingState, channels, isNewCategory} = this.props;
 
         if (!category) {
             return null;
         }
 
-        if (!isNewCategory && category.type !== CategoryTypes.DIRECT_MESSAGES && (!channels || !channels.length)) {
+        if (!isNewCategory && category.type !== CategoryTypes.DIRECT_MESSAGES && category.type !== CategoryTypes.CUSTOM && (!channels || !channels.length)) {
             return null;
         }
 
         const renderedChannels = channels.map(this.renderChannel);
 
         let categoryMenu: JSX.Element;
         let newLabel: JSX.Element;
-        let newDropBox: JSX.Element;
+        let newDropBox: (isDraggingOver: boolean) => JSX.Element;
         let directMessagesModalButton: JSX.Element;
         let hideArrow = false;
-        if (isNewCategory) {
+        if (isNewCategory && (!channels || !channels.length)) {
```

Comment:
We would then also need logic to reset it to new if a channel left, otherwise you have a blank category without a dropzone. I guess that's fine as well, but it's basically the same amount of code. This is just a shortcut. 1/5.

Reply:
Do we need to support that case? I wouldn't expect New to reappear if someone drags a channel into a category and then back out of it.

Reply:
1/5 I think it'd be worth it as a nicer UX. @andrewbrown00 any thoughts?

Reply:
Spoke with Harrison, we're good to keep this the way it is. Will add comments for clarity.